### PR TITLE
Avoid false positives in plugin search (installed tab)

### PR DESCRIPTION
**Pull request by zbynek**
See [JENKINS-65950](https://issues.jenkins.io/browse/JENKINS-65950).

Also: searching for "http" finds literally all the plugins, because the cells contain links. 

This PR reduces the search for plugins to the plugin ID and to the text content of the cell with description and name, avoiding matches in the HTML markup.

### Proposed changelog entries

* Avoid false positives in plugin search (installed tab).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [n/a] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by daniel-beck** [comment]
Does this change mean a substantially different display name will no longer be found? Would this no longer find `workflow-durable-task-step` when searching for "Nodes and Processes"?

Also, unsure how lowercasing would affect Turkish locales.

This looks like it also addresses the problem of discovering dependent/depending plugins.

**Comment by zbynek** [comment]
>  unsure how lowercasing would affect Turkish locales

The lowercase call was there before, so the situation for Turkish users didn't change. Since JS is handling this differently than Java (toLowerCase vs  toLocaleLowerCase), it should be fine in most cases (not sure about private plugins with actual turkish descrptions...).

> Would this no longer find workflow-durable-task-step when searching for "Nodes and Processes"?

The search now considers plugin ID and content of the `.details` cell (contians name and excerpt), so both `workflow` and `Nodes` should match.

**Review by oleg-nenashev — COMMENTED** [review]
I am not sure about that. Plugin IDs might be legacy, e.g. ones with "slave". Any chance we could just convert HTML to raw text inside Javascript?

**Comment by daniel-beck** [comment]
> The search now considers plugin ID and content of the `.details` cell (contians name and excerpt), so both `workflow` and `Nodes` should match.

Thanks! I missed that it's both name and description and not just the latter.

**Comment by zbynek** [comment]
@oleg-nenashev 

> Any chance we could just convert HTML to raw text inside Javascript?

That's what this PR is doing (innerText instead of innerHTML)

> Plugin IDs might be legacy

The IDs don't get more exposed than they are now. If a user types "ssh-slaves" I think it makes more sense to show SSH Build Agents then an empty search result. Apparently there are users who search by ID, e.g. the author of the original ticket, I didn't want to kill this feature.

File: core/src/main/resources/hudson/PluginManager/_table.js [hunk 1]
```diff
@@ -21,7 +21,8 @@ Behaviour.specify("#filter-box", '_table', 0, function(e) {
             }
             var makeVisible = true;
 
-            var content = items[i].innerHTML.toLowerCase();
+            var pluginId = items[i].getAttribute('data-plugin-id');
+            var content = (items[i].querySelector('.details').innerText + pluginId).toLowerCase();
```
Comment (daniel-beck) [inline core/src/main/resources/hudson/PluginManager/_table.js:25]:
```suggestion
            var content = (items[i].querySelector('.details').innerText + " " + pluginId).toLowerCase();
```

Currently, finding Structs Plugin when filtering for `objects.str` seems weird. Who knows what weird accidental matches are possible this way?

Untested.

**Review by daniel-beck — APPROVED** [review]

File: core/src/main/resources/hudson/PluginManager/_table.js (hunk 1 above)
Comment (daniel-beck) [reply core/src/main/resources/hudson/PluginManager/_table.js]: I was looking for `putti` (typoed "PuTTY", don't judge) and found Timestamper 😄 

> Screenshot

**Comment by zbynek** [comment]
@oleg-nenashev thanks for checking this. Do you consider your comment above blocking?

**Review by oleg-nenashev — APPROVED** [review]
Not at all. I just misinterpreted the behavior. Given the current explanation it looks perfectly fine with me

We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Comment by janfaracik** [comment]
Heads up that the Updates tab search field doesn't seem to work anymore - get this error in the console:

image

Running Safari 15.1, macOS Big Sur

**Pull request merged**